Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -185,6 +186,77 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::starts_with)
     return Value(string.substring(start, search_string_length) == search_string);
 }
 
+static bool is_regexp(VM& vm, GlobalObject& global_object, Value possible_regexp)
+{
+    if (!possible_regexp.is_object())
+        return false;
+
+    auto matcher = possible_regexp.as_object().get(vm.well_known_symbol_match());
+    if (vm.exception())
+        return false;
+    if (!matcher.is_empty() && !matcher.is_undefined()) {
+        return !matcher.to_boolean();
```

Comment:
When the spec says "If matcher is not undefined, return ! ToBoolean(matcher).", the "!" just means the operation cannot throw, and not exception check is needed (yes, it's a confusing way to express this):

https://tc39.es/ecma262/#sec-returnifabrupt-shorthands

> Similarly, prefix ! is used to indicate that the following invocation of an abstract or syntax-directed operation will never return an abrupt completion and that the resulting Completion Record's [[Value]] field should be used in place of the return value of the operation.

So:

```suggestion
        return matcher.to_boolean();
```